contracts/DonationCampaigns.sol

```diff
@@ -96,9 +245,17 @@ contract DonationCampaigns {
 		  "Withdrawal failed."
 		);
         campaigns[_index].donationWithdrawn = true;
+        emit DonationWithdrawn(_index, msg.sender, campaignDonation[_index]);
     }
 
+    /** @dev Make a donation to a campaign.
+      * @param _index The index of the campaign.
+      * @param donation The donation amount.
+    */
     function makeDonation(uint _index, uint donation) public payable {
+        require(_index < campaignsLength, "Invalid campaign index.");
+        require(donation > 0, "Donation amount must be greater than zero.");
+        require(msg.sender != campaigns[_index].owner, "Campaign creator cannot donate to their own campaign.");
```

Comment:
This is not needed

Reply:
Why do you think validations are not necessary